src/main/java/ru/mercuriev/game/of/life/graph/cell/CellBuilder.java

```diff
@@ -0,0 +1,179 @@
+package ru.mercuriev.game.of.life.graph.cell;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This stands for object which:
+ * 1. holds the pointer to the current cell ({@link Cell}) while constructing new Graph
+ * 2. appends new cells to the graph
+ * 3. merging together current and specified CellBuilder
+ * Each time any method of the object is called current cell is the right and bottom most cell in graph (*1*)
+ *
+ * Class is not thread safe - do not use it without proper synchronization (e.g. in parallel stream )
+ *
+ * @author paul
+ */
+final class CellBuilder {
+
+    /**
+     * current cell
+     */
+    private Cell cell = null;
+
+    public CellBuilder append(int value) {
+
+        if (cell == null) {
+            cell = new Cell(value);
+        } else {
+            Cell next = new Cell(value);
+            cell.right = next;
+            next.left = cell;
+            cell = cell.right;
+        }
+        return this;
+
+    }
+
+    /**
+     * merge this CellBuilder with another horizontally
+     * as a result - the most right cell of this builder is merged with the most left of the right builder
+     * @return this
+     * @throws NullPointerException if rightBuilder == null
+     */
+    public CellBuilder mergeHorizontal(CellBuilder rightBuilder) {
+
+        CellBuilder leftBuilder = this;
+        Objects.requireNonNull(rightBuilder);
+
+        if (rightBuilder.cell == null) {
+            return this; // no merge is needed
```

Comment:
No merge is needed because left builder value is not null and right builder value is null.